File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
               
        ListNode dummy = new ListNode();
		dummy.next = head;
        
        ListNode p1 = dummy;
        ListNode p2 = dummy;
        
        for(int i=0; i <= n;i++)
          p2 = p2.next;
        
        while(p2 !=null)
        {
            p2 = p2.next;
            p1 = p1.next;            
        }
        
        p1.next = p1.next.next;
        return dummy.next;
        
        
    }
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        
        ListNode curr = new ListNode(0);
        curr.next = head;
        
        ListNode p1 = curr;
        ListNode p2 = curr;
        
        for(int i=0;i <= n;i++){
            p1 = p1.next;
        }
        
        while(p1 != null)
        {
            p1 = p1.next;
            p2 = p2.next;
        }
        
        p2.next = p2.next.next;
        
        return curr.next;
        
    }
}